packages/x-date-pickers/src/DateField/tests/editing.DateField.test.tsx

```diff
@@ -516,6 +516,234 @@ describe('<DateField /> - Editing', () => {
+          expectedValue: '10',
+        });
+      });
+
+      it('should flip day field when value is lower than 5', () => {
+        testFieldKeyPress({
+          format: adapter.formats.dayOfMonth,
+          defaultValue: adapter.date('2022-01-04'),
+          key: 'PageDown',
+          expectedValue: '30',
+        });
+      });
+    });
+
+    describe('weekday section (PageDown)', () => {
+      it('should set weekday to Saturday when no value is provided', () => {
+        testFieldKeyPress({
+          format: adapter.formats.weekday,
+          key: 'PageDown',
+          expectedValue: 'Saturday',
+        });
+      });
+
+      it('should decrement weekday by 5 when value is provided', () => {
+        testFieldKeyPress({
+          format: adapter.formats.weekday,
+          defaultValue: adapter.date('2024-06-22'),
+          key: 'PageDown',
+          expectedValue: 'Monday',
+        });
+      });
+
+      it('should flip weekday field when value is lower than 5', () => {
+        testFieldKeyPress({
+          format: adapter.formats.weekday,
+          defaultValue: adapter.date('2024-06-23'),
+          key: 'PageDown',
+          expectedValue: 'Tuesday',
+        });
+      });
+    });
+
+    describe('month section (PageDown)', () => {
+      it('should set month to December when no value is provided', () => {
+        testFieldKeyPress({
+          format: adapter.formats.month,
+          key: 'PageDown',
+          expectedValue: 'December',
+        });
+      });
+
+      it('should decrement month by 5 when value is provided', () => {
+        testFieldKeyPress({
+          format: adapter.formats.month,
+          defaultValue: adapter.date('2022-10-15'),
+          key: 'PageDown',
+          expectedValue: 'May',
+        });
+      });
+
+      it('should flip month field when value is lower than 5', () => {
+        testFieldKeyPress({
+          format: adapter.formats.month,
+          defaultValue: adapter.date('2022-04-15'),
+          key: 'PageDown',
+          expectedValue: 'November',
+        });
+      });
+    });
+
+    describe('year section (PageDown)', () => {
+      it('should set year to current year when no value is provided', () => {
+        testFieldKeyPress({
+          format: adapter.formats.year,
+          key: 'PageDown',
+          expectedValue: new Date().getFullYear().toString(),
+        });
+      });
+
+      it('should decrement year by 5 when value is provided', () => {
+        testFieldKeyPress({
+          format: adapter.formats.year,
+          defaultValue: adapter.date('2022-01-15'),
+          key: 'PageDown',
+          expectedValue: '2017',
+        });
+      });
+
+      it('should flip year field when value is lower than 5', () => {
+        testFieldKeyPress({
+          format: adapter.formats.year,
+          defaultValue: adapter.date('0003-01-15'),
+          key: 'PageDown',
+          expectedValue: adapter.lib === 'dayjs' ? '1898' : '9998',
```

Comment:
@flaviendelangle WDYT about making the `minimum` and `maximum` aware of the `minDate` and `maxDate`?
Didn't we do it only, because `shouldDisableYear` exists? 🤔 
IMHO, the current behavior seems a bit strange, especially given the differences in adapters. 🙈